src/Grpc.Net.Client/Internal/Compression/GzipCompressionProvider.cs

```diff
@@ -0,0 +1,66 @@
+﻿#region Copyright notice and license
```

Comment:
This is the same as the GzipCompressionProvider in Grpc.AspNetCore.Server. Can we create a shared folder and add it to the Compile itemgroup instead?

Reply:
This is internal, the server is not. I might make the server compression API internal, but unifying code will bloat this PR. Will do it separately.